packages/core/src/js/tracing/reactnativetracing.ts

```diff
@@ -97,6 +98,22 @@ export const reactNativeTracingIntegration = (
     idleTimeoutMs: options.idleTimeoutMs ?? defaultIdleOptions.idleTimeout,
   };
 
+  const userShouldCreateSpanForRequest = finalOptions.shouldCreateSpanForRequest;
+
+  // Drop Dev Server Spans
+  const devServerUrl = getDevServer()?.url || '';
+  const finalShouldCreateSpanForRequest = (url: string): boolean => {
+    if (url.includes(devServerUrl)) {
+      return false;
+    }
+    if (userShouldCreateSpanForRequest) {
+      return userShouldCreateSpanForRequest(url);
+    }
+    return true;
+  };
```

Comment:
i: We could avoid this logic and rely only on the customer logic when devserver isn't found.
This can also avoid issues with `url.includes(''))` that can return true to any kind of string
```suggestion
  const devServerUrl = getDevServer()?.url;
  const finalShouldCreateSpanForRequest =
    devServerUrl === undefined
      ? userShouldCreateSpanForRequest
      : (url: string): boolean => {
          if (url.includes(devServerUrl)) {
            return false;
          }
          if (userShouldCreateSpanForRequest) {
            return userShouldCreateSpanForRequest(url);
          }
          return true;
        };
```

Reply:
Thank you for the suggestion @lucas-zimerman and for catching this case 🙇